flink-runtime/src/main/java/org/apache/flink/runtime/io/network/buffer/LocalBufferPool.java

```diff
@@ -355,6 +359,12 @@ public void lazyDestroy() {
 					listener.notifyBufferDestroyed();
 				}
 
+				// notify the potential blocking request thread
+				final CompletableFuture<?> toNotify = availabilityHelper.isAvailable();
+				if (toNotify != AVAILABLE && !toNotify.isDone()) {
+					toNotify.complete(null);
```

Comment:
nit: it is better to move `toNotify.complete(null)` out of `sync` like the other places did.